tensorflow_addons/losses/crf.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implementing Conditional Random Field loss."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+
+from tensorflow_addons.layers.crf import CRF
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class ConditionalRandomFieldLoss(object):
```

Comment:
Why not extend `tf.keras.losses.Loss`, and implement `call`?

Reply:
I add a doc (`design_docs/crf.md`) to explain why loss function cannot be implemented as a subclass of `tf.keras.losses.Loss`.  If there is an error, please correct me.